[Java] Configuration package names are not sanitized

I am working with a pipeline that generates the swagger-codegen config json file. Sometimes, a json file is generated that contains dashes in the apiPackage parameter. Dashes are not valid values in java package names, so it would be nice if swagger-codegen would sanitize this or throw an error.

Here is an example/repro (requires mvn, java) :

**1. Make a temporary working directory**
```
cd `mktemp -d /tmp/tmp.XXXXXXX`
```

**2. Download swagger-codegen-cli.jar**
```
mvn org.apache.maven.plugins:maven-dependency-plugin:2.8:get  -DremoteRepositories=https://oss.sonatype.org/content/repositories/snapshots -Dartifact=io.swagger:swagger-codegen-cli:LATEST -Ddest=swagger-codegen-cli.jar
```

**3. Create a config with invalid values**
Note that there is a dash in the `apiPackage`
```
echo '{"apiPackage": "io.swagger.pet-store"}' > config.json
```

**4. Invoke swagger-codegen-cli.jar**
```
java -jar swagger-codegen-cli.jar generate -l java -c config.json -i http://petstore.swagger.io/v2/swagger.json
```

**5. Compile the generated output**
```
mvn compile;
```

Result:
```
...
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:2.3.2:compile (default-compile) on project swagger-java-client: Compilation failure: Compilation failure:
[ERROR] /tmp/tmp.1CUdUg0/src/main/java/io/swagger/pet-store/UserApi.java:[1,22] error: ';' expected
[ERROR] /tmp/tmp.1CUdUg0/src/main/java/io/swagger/pet-store/StoreApi.java:[1,22] error: ';' expected
[ERROR] /tmp/tmp.1CUdUg0/src/main/java/io/swagger/pet-store/PetApi.java:[1,22] error: ';' expected
...
```

--- a/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaClientCodegen.java
+++ b/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaClientCodegen.java
@@ -92,7 +92,7 @@ public class JavaClientCodegen extends DefaultCodegen implements CodegenConfig {
     @Override
     public void processOpts() {
         super.processOpts();
-
+        
         if (additionalProperties.containsKey("invokerPackage")) {
             this.setInvokerPackage((String) additionalProperties.get("invokerPackage"));
         } else {
@@ -130,6 +130,8 @@ public class JavaClientCodegen extends DefaultCodegen implements CodegenConfig {
             this.setLocalVariablePrefix((String) additionalProperties.get("localVariablePrefix"));
         }
 
+        this.sanitizeConfig();
+
         final String invokerFolder = (sourceFolder + File.separator + invokerPackage).replace(".", File.separator);
         supportingFiles.add(new SupportingFile("pom.mustache", "", "pom.xml"));
         supportingFiles.add(new SupportingFile("ApiClient.mustache", invokerFolder, "ApiClient.java"));
@@ -146,7 +148,26 @@ public class JavaClientCodegen extends DefaultCodegen implements CodegenConfig {
         supportingFiles.add(new SupportingFile("auth/OAuth.mustache", authFolder, "OAuth.java"));
     }
 
-
+    private void sanitizeConfig() {
+        // Sanitize any config options here. We also have to update the additionalProperties because 
+        // the whole additionalProperties object is injected into the main object passed to the mustache layer
+        
+        this.setApiPackage(sanitizePackageName(apiPackage));
+        if (additionalProperties.containsKey("apiPackage")) {
+            this.additionalProperties.put("apiPackage", apiPackage);
+        }
+        
+        this.setModelPackage(sanitizePackageName(modelPackage));
+        if (additionalProperties.containsKey("modelPackage")) {
+            this.additionalProperties.put("modelPackage", modelPackage);
+        }
+        
+        this.setInvokerPackage(sanitizePackageName(invokerPackage));
+        if (additionalProperties.containsKey("invokerPackage")) {
+            this.additionalProperties.put("invokerPackage", invokerPackage);
+        }
+    }
+    
     @Override
     public String escapeReservedWord(String name) {
         return "_" + name;
@@ -342,4 +363,14 @@ public class JavaClientCodegen extends DefaultCodegen implements CodegenConfig {
     public void setLocalVariablePrefix(String localVariablePrefix) {
         this.localVariablePrefix = localVariablePrefix;
     }
+
+    private String sanitizePackageName(String packageName) {
+        packageName = packageName.trim();
+        packageName = packageName.replaceAll("[^a-zA-Z0-9_\\.]", "_");
+        if(Strings.isNullOrEmpty(packageName)) {
+            return "invalidPackageName";
+        }
+        return packageName;
+    }
+
 }

